### ISSUE TITLE
Suggested improvements for "Variable substitution" section

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/docs#1780]
This is a very key section, and I wonder if it's worth spending the time to do another pass on.

Some suggestions:

1)  Introduce an anchor-type link directly to the "Variable substitution" section.

2)  Perhaps add another word like "ordering"/"precedence" to bubble up the purpose into the heading, e.g. "Variable substitution - ordering".

3) For "6.variables declared on the command line" this should include an example.  I'm afraid this could be confused with a variable set via:  `mvn -Dliberty.var.derby.url`  which of course maps to 5., not 6.   (I don't know the syntax or would list an example :) )

4) Maybe tweak the names to be more consistent:
```
TODAY:
   1. server.xml default variable values
   5. server.xml config
PROPOSED:
   1. server.xml variable defaultValue(s)
   5. server.xml variable values

As-is the connection between the two isn't too clear, and I think this is better.
```

**Comment by Charlotte-Holt** [comment]
@scottkurz Thanks for the suggestions!

1. I added an anchor link to the **Variable substitution precedence** section in [the overview doc that's going to be in the Basics section of the nav](https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/server-config-feature-intro.html). 
2. I updated the section title to **Variable substitution precedence** in https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/reference/config/server-configuration-overview.html.
3. Still need to work this out and get an example from a dev. @cbridgha, see Scott's suggestion number 3 above. Can you provide an example of "variables declared on the command line", as it's mentioned in the **Variable substituion precedence** section of https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/reference/config/server-configuration-overview.html? 
4. I updated the names as you suggested in https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/reference/config/server-configuration-overview.html.

**Comment by scottkurz** [comment]
Thanks @Charlotte-Holt.  I like it.. it looks good, and like the new links !

I found an example of "variables declared on the command line" here: 
https://openliberty.io/docs/20.0.0.9/reference/command/server-run.html#_usage_examples

It's: `server run myserver -- --variable_name=variable_value`

--- 

Minor comment, just noticing that it reads: 

     1. server.xml file `defaultValues`

then later in the text:

    Default values, which are specified in the server.xml file.

I could see changing to "server.xml file default values" to be more consistent though clear enough as-is.

**Comment by Charlotte-Holt** [comment]
@scottkurz Thanks for pointing out that example! I added it to the doc after, "You can also specify variables at startup from the command line. If you do, the variables that are specified on the command line override all other sources of variables and can’t be changed after the server starts:". I wasn't sure that the first `--` should be in the command, so I'm looking into that. If not, I'll update the command doc as well. 

For the default variables language, I updated these to "default variable values in the `server.xml` file" and "variables declared on the command line" in the list of precedence. I was trying to keep the language about default variables consistent between the list and the paragraph that begins "Default values, which are specified in the server.xml file...". Does this seem right to you, or did you prefer the list with `defaultValues`? 

Link: https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/reference/config/server-configuration-overview.html

**Comment by scottkurz** [comment]
@Charlotte-Holt .. 

1. I like this newer rendition better

    > default variable values in the server.xml file

though.. and this may seem like a nit, I'd instead write:

    > variable default values in the server.xml file

Maybe in plain English the first is better but I like keeping "default" and "values" close, not separate, since the attribute after all is "defaultValue".   


2. Just to make it even more obvious, could we take:
> To resolve variable references to their values, the following sources are consulted in order: 

and add:

> To resolve variable references to their values, the following sources are consulted in order, in increasing order of precedence, with later options superseding and taking precedence over earlier options.

Maybe I'm going a bit overboard.. but is there a way to spell it out just a bit more?


---
3. Pretty sure  from https://openliberty.io/docs/20.0.0.9/reference/command/server-run.html#_usage_examples  it is one "--" just to separate the list of variables ,  so like:

   > server start myServer --  --abc=123

But great if you can confirm since I didn't actually try this.

Thanks

**Comment by Charlotte-Holt** [comment]
@scottkurz Thx for the feedback! Updated in https://draft-openlibertyio.mybluemix.net/docs/20.0.0.12/reference/config/server-configuration-overview.html#variable-substitution. See what you think, and let me know if you spot anything else that needs updating.

**Comment by scottkurz** [comment]
@Charlotte-Holt  - I like it ! Thanks for being willing to keep revising this .

Never knew that's how to spell "supersede"  either :)

**Comment by Charlotte-Holt** [comment]
@scottkurz Of course. Thanks for your feedback and helping improve the docs! :) I'll move this into the **Ready to Publish** column. These updates are related to #3192.

**Comment by Charlotte-Holt** [comment]
Merged to `vNext` in #3391.

**Comment by Charlotte-Holt** [comment]
Merged to v20.0.0.12 in #3390. Closing this issue.
